tests/jobscript/19-exit-script.t

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2018 NIWA & British Crown (Met Office) & Contributors.
+# 
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#------------------------------------------------------------------------------
+# Test exit-script.
+
+. "$(dirname "${0}")/test_header"
+set_test_number 9
+
+install_suite "${TEST_NAME_BASE}" "${TEST_NAME_BASE}"
+
+run_ok "${TEST_NAME_BASE}-validate" cylc validate "${SUITE_NAME}"
+
+# 1) runs on normal successful job exit?
+run_ok "${TEST_NAME_BASE}-run" \
+  cylc run --debug --no-detach "${SUITE_NAME}" 
+grep_ok 'Cheesy peas!' "${SUITE_RUN_DIR}/log/job/1/foo/01/job.out"
+grep_fail 'Oops!' "${SUITE_RUN_DIR}/log/job/1/foo/01/job.err"
+
+# 2) runs on internal early EXIT?
+run_fail "${TEST_NAME_BASE}-run" \
+  cylc run --debug --no-detach --set=EXIT=true "${SUITE_NAME}" 
+grep_ok 'Cheesy peas!' "${SUITE_RUN_DIR}/log/job/1/foo/01/job.out"
+grep_ok 'EXIT Oops!' "${SUITE_RUN_DIR}/log/job/1/foo/01/job.err"
+
+# 3) runs on external job TERM?
+cylc run --set=NSLEEP=30 "${SUITE_NAME}"
+sleep 3
+PID_SET=$(grep CYLC_JOB_PID "${SUITE_RUN_DIR}/log/job/1/foo/01/job.status")
+eval $PID_SET
```

Comment:
Can avoid `eval` with one of these:

```sh
CYLC_JOB_PID=$(awk -F= '$1 == "CYLC_JOB_PID" {print $2}' "${SUITE_RUN_DIR}/log/job/1/foo/01/job.status")
# or
CYLC_JOB_PID=$(sed -n 's/^CYLC_JOB_PID=//p' "${SUITE_RUN_DIR}/log/job/1/foo/01/job.status")
```

Reply:
Yeah, but look at how nice and clean `eval $PID_SET` is compared to those.  Is `eval` really evil in this context?

Reply:
Only one letter away!

Reply:
The simple `sed` statement, for example, will replace both the `grep` and the `eval` lines - and also allow the reader to have a better understanding of where the `CYLC_JOB_PID` variable comes from when it is used in the following line.

Reply:
OK, I do agree with that, I'll make a quick change. I was making direct use of the job status file syntax, but actually it's not a shell script and could in principle change in future ... which would break my code (well, it would break yours too, but more transparently!).

Reply:
done